### PR TITLE
Fetch 'info' dict from API instead of scraping

### PR DESCRIPTION
**Pull request by qianyun210603**
Due to the persistent failure of info property of Tickers due to decryption error of yahoo finance web pages, I added a failback to get some information from unofficial api https://query1.finance.yahoo.com/v7/finance/quote. which can get the ticker short/long name, listed date, exchange, timezone etc.

Quick test
```
Type "help", "copyright", "credits" or "license" for more information.
>>> import yfinance as yf
>>> ticker = yf.Ticker("KWEB")
>>> ticker.history()
                                Open       High        Low      Close    Volume  Dividends  Stock Splits  Capital Gains
Date
2023-02-23 00:00:00-05:00  31.150000  31.170000  29.785000  30.030001  21059600        0.0           0.0            0.0
2023-02-24 00:00:00-05:00  29.040001  29.350000  28.670000  28.910000  24900400        0.0           0.0            0.0
2023-02-27 00:00:00-05:00  29.559999  29.674999  29.230000  29.379999  16049700        0.0           0.0            0.0
2023-02-28 00:00:00-05:00  29.100000  29.620001  28.990000  29.190001  14846200        0.0           0.0            0.0
2023-03-01 00:00:00-05:00  30.790001  30.940001  30.344000  30.500000  26761200        0.0           0.0            0.0
2023-03-02 00:00:00-05:00  30.379999  31.545000  30.260000  31.490000  25618400        0.0           0.0            0.0
2023-03-03 00:00:00-05:00  31.420000  31.785000  31.420000  31.650000  15196500        0.0           0.0            0.0
2023-03-06 00:00:00-05:00  31.370001  31.469999  30.915001  30.959999  15815100        0.0           0.0            0.0
2023-03-07 00:00:00-05:00  30.570000  30.590000  29.920000  30.049999  17186600        0.0           0.0            0.0
2023-03-08 00:00:00-05:00  29.570000  29.834999  29.350000  29.740000  14574100        0.0           0.0            0.0
2023-03-09 00:00:00-05:00  28.950001  29.010000  28.030001  28.110001  29801100        0.0           0.0            0.0
2023-03-10 00:00:00-05:00  28.080000  28.615999  27.889999  28.219999  21822800        0.0           0.0            0.0
2023-03-13 00:00:00-04:00  28.190001  29.010000  28.059999  28.670000  21528700        0.0           0.0            0.0
2023-03-14 00:00:00-04:00  28.510000  28.980000  28.309999  28.930000  16422800        0.0           0.0            0.0
2023-03-15 00:00:00-04:00  28.180000  28.400000  27.684999  28.180000  22224000        0.0           0.0            0.0
2023-03-16 00:00:00-04:00  28.080000  29.020000  28.010000  28.950001  18518400        0.0           0.0            0.0
2023-03-17 00:00:00-04:00  29.330000  29.500000  28.639999  28.940001  19655600        0.0           0.0            0.0
2023-03-20 00:00:00-04:00  28.340000  29.115000  27.950001  28.709999  16119100        0.0           0.0            0.0
2023-03-21 00:00:00-04:00  29.049999  29.320000  28.790001  29.139999  13043800        0.0           0.0            0.0
2023-03-22 00:00:00-04:00  29.469999  29.500000  28.900000  28.920000  10900300        0.0           0.0            0.0
>>> ticker.info
{'symbol': 'KWEB', 'dividendDate': 1513900800, 'twoHundredDayAverageChangePercent': -0.0001659751, 'trailingThreeMonthNavReturns': -5.51232, 'fiftyTwoWeekLowChangePercent': 0.6794426, 'language': 'en-US', 'regularMarketDayRange': '28.9 - 29.5', 'regularMarketDayHigh': 29.5, 'twoHundredDayAverageChange': -0.0048007965, 'askSize': 280, 'twoHundredDayAverage': 28.9248, 'fiftyTwoWeekHighChange': -7.2699986, 'fiftyTwoWeekRange': '17.22 - 36.19', 'fiftyDayAverageChange': -3.018999, 'averageDailyVolume3Month': 18945966, 'exchangeDataDelayedBy': 0, 'firstTradeDateMilliseconds': Timestamp('2013-08-01 09:30:00-0400', tz='America/New_York'), 'trailingAnnualDividendRate': 0.0, 'fiftyTwoWeekLow': 17.22, 'market': 'us_market', 'regularMarketVolume': 10891543, 'postMarketPrice': 28.86, 'quoteSourceName': 'Nasdaq Real Time Price', 'messageBoardId': 'finmb_243136794', 'priceHint': 2, 'exchange': 'PCX', 'regularMarketDayLow': 28.9, 'ytdReturn': -3.50993, 'sourceInterval': 15, 'shortName': 'KraneShares Trust KraneShares C', 'region': 'US', 'fiftyDayAverageChangePercent': -0.094523914, 'fullExchangeName': 'NYSEArca', 'gmtOffSetMilliseconds': -14400000, 'regularMarketOpen': 29.47, 'regularMarketTime': '4:00PM EDT', 'regularMarketChangePercent': -0.7549736, 'quoteType': 'ETF', 'trailingAnnualDividendYield': 0.0, 'fiftyTwoWeekLowChange': 11.700001, 'averageDailyVolume10Day': 19003660, 'fiftyTwoWeekHighChangePercent': -0.2008842, 'typeDisp': 'ETF', 'trailingPE': 20.814081, 'tradeable': False, 'postMarketTime': '7:59PM EDT', 'currency': 'USD', 'fiftyTwoWeekHigh': 36.19, 'regularMarketPreviousClose': 29.14, 'postMarketChangePercent': -0.207467, 'trailingThreeMonthReturns': -5.51232, 'exchangeTimezoneName': 'America/New_York', 'regularMarketChange': -0.21999931, 'bidSize': 32, 'cryptoTradeable': False, 'fiftyDayAverage': 31.939, 'exchangeTimezoneShortName': 'EDT', 'marketState': 'POSTPOST', 'customPriceAlertConfidence': 'HIGH', 'regularMarketPrice': 28.92, 'postMarketChange': -0.0599995, 'ask': 29.09, 'epsTrailingTwelveMonths': 1.389444, 'bid': 28.84, 'triggerable': True, 'longName': 'KraneShares CSI China Internet ETF'}
>>> exit()

```

### CODE REVIEW

File: yfinance/base.py [hunk 1]
```diff
@@ -1910,3 +1917,25 @@ def get_history_metadata(self) -> dict:
...
+                    query1_info[k] = v["fmt"] if k in {"regularMarketTime", "postMarketTime"} else v["raw"]
+                elif isinstance(v, str):
+                    query1_info[k] = v.replace("\xa0", " ")
+            if "firstTradeDateMilliseconds" in query1_info:
+                query1_info["firstTradeDateMilliseconds"] = _pd.Timestamp(
+                    query1_info["firstTradeDateMilliseconds"], unit="ms", tz="UTC"
+                ).tz_convert(query1_info.get("exchangeTimezoneName", "UTC"))
+            self._query1_info = query1_info
+
+        return self._query1_info
```
Comment (ValueRaider) [inline yfinance/base.py:1941]: Move this into yfinance/scrapers/quote.py

File: yfinance/base.py [hunk 2]
```diff
@@ -1471,8 +1473,13 @@ def get_mutualfund_holders(self, proxy=None, as_dict=False):
...
     def get_info(self, proxy=None) -> dict:
-        self._quote.proxy = proxy
-        data = self._quote.info
+        try:
+            self._quote.proxy = proxy
+            data = self._quote.info
+        except Exception:
+            data = {}
+        data1 = self.get_query1_info()
+        data.update(data1)
```
Comment (ValueRaider) [inline yfinance/base.py:1482]: Revert this change. Instead, modify yfinance/scrapers/quote.py::`Quote.info()` to call your new method instead of `_scrape()`

**Review by ValueRaider — COMMENTED** [review]
Great work. I made some comments on moving new code to better places.

File: yfinance/base.py (hunk 2 above)
Comment (qianyun210603) [reply yfinance/base.py]: modified as suggested.

File: yfinance/scrapers/quote.py [hunk 3]
```diff
@@ -236,12 +240,36 @@ def _scrape(self, proxy):
...
+        )
+        query1_info = next(
+            (info for info in result.get("quoteResponse", {}).get("result", []) if info["symbol"] == self._data.ticker),
+            None,
+        )
+        for k, v in query1_info.items():
+            if isinstance(v, dict) and "raw" in v and "fmt" in v:
+                query1_info[k] = v["fmt"] if k in {"regularMarketTime", "postMarketTime"} else v["raw"]
+            elif isinstance(v, str):
+                query1_info[k] = v.replace("\xa0", " ")
```
Comment (ValueRaider) [inline yfinance/scrapers/quote.py:259]: Add:
```
elif isinstance(v, (int, bool)):
  query1_info[k] = v
```

Then don't need special handling for "firstTradeDateMilliseconds"

Comment (ValueRaider) [reply yfinance/scrapers/quote.py:259]: I've done this now

**Comment by qianyun210603** [comment]
@ValueRaider yes, I'm fine with the change.

But seems I cannot operate approve on my own pr. maybe you can just go forward to merge.

Thanks!

**Comment by iukea1** [comment]
Not all heroes wear capes....
 But you should
